app/src/main/java/org/oppia/app/player/audio/AudioViewModel.kt

```diff
@@ -1,17 +1,118 @@
 ) : ViewModel() {
-  val currentLanguageCode = ObservableField<String>("en")
 
+  private lateinit var explorationId: String
+  private var voiceoverMap = mapOf<String, Voiceover>()
+
+  /** Mirrors PlayStatus in AudioPlayerController except adds LOADING state */
+  enum class AudioPlayStatus {
+    LOADING,
+    PREPARED,
+    PLAYING,
+    PAUSED,
+    COMPLETED
+  }
+
+  val currentLanguageCode = ObservableField<String>()
+
+  val durationLiveData: LiveData<Int> by lazy {
+    processDurationLiveData()
+  }
+  val positionLiveData: LiveData<Int> by lazy {
+    processPositionLiveData()
+  }
+  val playStatusLiveData: LiveData<AudioPlayStatus> by lazy {
+    processPlayStatusLiveData()
+  }
+
+  fun setVoiceoverMappings(map : Map<String, Voiceover>) {
+    voiceoverMap = map
+  }
+
+  fun setExplorationId(id: String) {
+    explorationId = id
+  }
+
+  /** Sets language code for data binding and changes data source to correct audio */
   fun setAudioLanguageCode(languageCode: String) {
     currentLanguageCode.set(languageCode)
+    audioPlayerController.changeDataSource(voiceOverToUri(voiceoverMap[languageCode]))
+  }
+
+  /** Plays or pauses AudioController depending on passed in state */
+  fun togglePlayPause(type: AudioPlayStatus?) {
+    if (type == AudioPlayStatus.PLAYING) {
+      audioPlayerController.pause()
+    } else {
+      audioPlayerController.play()
+    }
+  }
+
+
+  fun pauseAudio() = audioPlayerController.pause()
+  fun handleSeekTo(position: Int) = audioPlayerController.seekTo(position)
+  fun handleRelease() = audioPlayerController.releaseMediaPlayer()
+
+  private val playProgressResultLiveData: LiveData<AsyncResult<PlayProgress>> by lazy {
+    audioPlayerController.initializeMediaPlayer()
+  }
+
+  private fun processDurationLiveData(): LiveData<Int> {
+    return Transformations.map(playProgressResultLiveData, ::processDurationResultLiveData)
+  }
+
+  private fun processPositionLiveData(): LiveData<Int> {
+    return Transformations.map(playProgressResultLiveData, ::processPositionResultLiveData)
+  }
+
+  private fun processPlayStatusLiveData(): LiveData<AudioPlayStatus> {
+    return Transformations.map(playProgressResultLiveData, ::processPlayStatusResultLiveData)
+  }
+
+  private fun processDurationResultLiveData(playProgressResult: AsyncResult<PlayProgress>): Int {
+    if (playProgressResult.isPending()) {
+      return 0
+    }
+    return playProgressResult.getOrThrow().duration
+  }
+
+  private fun processPositionResultLiveData(playProgressResult: AsyncResult<PlayProgress>): Int {
+    if (playProgressResult.isPending()) {
+      return 0
+    }
+    return playProgressResult.getOrThrow().position
+  }
+
+  private fun processPlayStatusResultLiveData(playProgressResult: AsyncResult<PlayProgress>): AudioPlayStatus {
+    if (playProgressResult.isPending()) return AudioPlayStatus.LOADING
+    return when (playProgressResult.getOrThrow().type) {
+      PlayStatus.PREPARED -> AudioPlayStatus.PREPARED
```

Comment:
Consider UiAudioPlayStatus instead of just AudioPlayStatus. That might help further differentiate from PlayStatus.